cc/constraints/BUILD.bazel

```diff
@@ -0,0 +1,8 @@
+package(default_visibility = ["//visibility:public"])
+
+constraint_setting(name = "compiler")
+
+constraint_value(
+    name = "gcc-6",
+    constraint_setting = ":compiler",
```

Comment:
Does this override a standard setting that Bazel uses to select a compiler?  Is specifying `gcc-6` enough to allow Bazel to locate all associated tools like `g++-6`?

Reply:
I guess just pointing me at the docs where that explains this would work too :-D

Reply:
It doesn't override any standard Bazel settings. Out of the box this attribute alone won't have any affect on the toolchain that is selected.

You need to pair it with a custom `platform` target and `toolchain` target for that.

In the example above I provided the minimal amount of code necessary to trigger this condition so that it "activates" this constraint.

In Bazel a `platform` basically defines a bundle of `constraint_values`. When the platform is activated via `--platform`, those constraint values are active.

Bazel selects the first toolchain that it knows about that matches all of the active `constraint_values`. In the example above the platform I defined is the minimum set of constrain values to "match" the default cc toolchain, plus our own custom constraint setting that we defined here. I still had to specify the gcc-6 toolchain using `CC` and `CXX`.

So how I imagine this being used is we might have a custom `debian_stretch` platform that defines some constraint settings, and a corresponding `gcc-6` toolchain that is selected when that platform is activated.

We could also end up supporting another target platform, `foo`, that corresponds to a yocto toolchain or something that also uses gcc-6.

In both cases the build system will still choose the right set of flags for the compile options.

Reply:
Here's the official documentation on `constraints` and `platforms`: https://bazel.build/extending/platforms#constraints-platforms

Reply:
Cool, thanks